x-pack/plugins/xpack_main/server/lib/populate_ui_capabilities.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { UICapabilities } from 'ui/capabilities';
+import { Feature } from '../../types';
+
+export function populateUICapabilities(
+  xpackMainPlugin: Record<string, any>,
+  uiCapabilities: UICapabilities
+): UICapabilities {
+  const features: Feature[] = xpackMainPlugin.getFeatures();
+
+  const featureCapabilities: UICapabilities[] = features.map(getCapabilitiesFromFeature);
+
+  return mergeCapabilities(uiCapabilities || {}, ...featureCapabilities);
+}
+
+function getCapabilitiesFromFeature(feature: Feature): UICapabilities {
+  const capabilities: UICapabilities = {
+    navLinks: {},
```

Comment:
What's the significance of including `navLinks` here? When I initially saw this here, I thought that we'd end up overwriting the navLinks with this empty object, but the order of the arguments in the merge prevents this.

Reply:
The only purpose it serves is to satisfy the `UICapabilities` interface, as `navLinks` is marked as a required property there. I could update this to return `Record<string, Record<string, boolean>>` if you'd prefer

Reply:
Ah... yeah, using a different type here seems to make sense, as we'd have to do the same with the `management` section which seems counter-intuitive.

Reply:
Resolved via https://github.com/elastic/kibana/pull/27031/commits/99fb34bc042fb3888270249a7de417a0d5a2a235